### ISSUE TITLE
XYPlot size changes when panning

### ISSUE THREAD

**Issue by gsteckman** [KoalaPlot/koalaplot-core#101]
XYPlot sizes itself to maximize the plot area, leaving room for the axis tick labels. When panning, e.g. horizontally to the left, the plot area will expand to the right and then snap back when the next major tick label comes into view.

**Comment by andrey-ananiev** [comment]
I don't really understand why this happens. I will look into it and look for a solution.
Previously I used the VICO library to draw graphs. In it, panning and zooming worked correctly and smoothly. Maybe we can see this implementation in it? For example, [PointerInputScope.detectZoomGestures](https://github.com/patrykandpatrick/vico/blob/master/vico/compose/src/main/java/com/patrykandpatrick/vico/compose/common/ZoomGestures.kt)

**Comment by gsteckman** [comment]
As an example, assume we have a plot with an x-axis range from 1...100 and the min view range is 1...10. Initially the plot is drawn from 1...100. The right most major tick will be 100 and it will be at the right most edge of the plot drawing area. The sizing algorithm first estimates the size of all the x-axis tick labels, and it will measure the label "100" as the last major tick label. Since it needs to put the text "100" centered on a vertical line that forms the right edge of the plot area, that right edge cannot be flush right with the screen, instead it must be moved in to the left to account for half the width of "100".

Now imagine the user zooms in to display the range 80...90. The same measurement will occur, but now with "90" as the right most major tick label. Because "90" is narrower than "100", the plot area will grow in width a little bit, because not as much space is needed at the right edge for the label and so the last major tick can be a little closer to the right edge of the screen.

Next, suppose the user pans so the plot range is from 80.5...90.5. Now the last major tick "90" no longer needs to be at the rightmost part of the plot, it needs to be moved to the left just a little bit. Because the algorithm is maximizing the amount of space given to the plot, it can now put "90" flush with the right edge of the screen, and use the space between the center of the "90" and the right edge of the screen to draw the plot content from 90 to 90.5. This expansion of the plot width will be visible to the user, and as the user continues to pan and eventually reaches the point where the last major tick is "91", the plot width will snap back to where the right edge of the plot area is centered on the "91".

There was no zoom/pan feature when I wrote this approach to size and maximize the plot drawing area, so I didn't consider this situation. One solution is to pre-measure all the possible major tick labels across the whole axis range, and use the maximum width as a constant to be used for all graph sizing calculations. This will likely take some changes to the AxisModel as it currently only produces major tick values for what is in the current view, not for the entire range of the axis, which would be required to generate (Compose) all labels up front for measurement. The downside to this is that it will be causing labels to be Composed which might not get drawn on the screen, so it's not 100% efficient. But at the moment I don't know of another way to get the size of the labels before they might be needed.

**Comment by andrey-ananiev** [comment]
Why not stop measuring the labels and focus on displaying the graph itself. Let the labels that do not fit in width either overlap the adjacent ones or are cut off?

**Comment by gsteckman** [comment]
Yes, that's an option. I was playing around with the amCharts demos and it looks like they do it that way.

**Comment by andrey-ananiev** [comment]
I will be very grateful if you can solve the problem with drawing X-axis labels. Thank you very much for your support and detailed explanations.

**Comment by andrey-ananiev** [comment]
I applied a temporary solution to the problem. It is as follows:
1. Disable X-axis label
2. Increase the range of the Y axis from the bottom side
3. Add XYAnnotation in XYGraph Content

```
model.xData.forEach { x ->
            XYAnnotation(
                location = Point(x, yMin),
                anchorPoint = AnchorPoint.TopCenter
            ) {
                AxisLabel(
                    modifier = Modifier.padding(vertical = 8.dp),
                    label = when (x) {
                        is YearMonth -> x.toMonthYearName(resourceManager, isTwoLine = true)
                        is LocalDate -> x.dayOfMonth.toString()
                        else -> x.toString()
                    }
                )
            }
        }
```
